build_tools/release_automation.py

```diff
@@ -0,0 +1,262 @@
+        {'name': 'Rozycko, Piotr','affiliation': 'Data Management and Software Centre, European Spallation Source ERIC' },
+        {'name': 'Snow, Tim','affiliation': 'Diamond Light Source','orcid': '0000-0001-7146-6885'},
+        {'name': 'Washington, Adam','affiliation': 'STFC - Rutherford Appleton Laboratory'}
+        ],
+    'grants': [{'id': '10.13039/501100000780::654000'}],
+    'license': 'BSD-3-Clause',
+    'upload_type': 'software',
+    'access_right': 'open',
+    'communities': [{'identifier': 'ecfunded'}, {'identifier': 'zenodo'}],
+    'prereserve_doi': 'true'
+    }
+}
+
+class SplitArgs(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, values.split(','))
+
+def generate_zenodo(sasview_data, zenodo_api_key):
+    """
+    Generating zenodo record
+    :return:
+    """
+    #get list of existing depositions
+    r = requests.get(zenodo_url+"/api/deposit/depositions",
+        params={'access_token':zenodo_api_key})
+    if r.status_code != 200:
+        print("Failure of zenodo connection check. Terminating.")
+        sys.exit(1)
+
+
+    #create empty upload
+    headers = {"Content-Type": "application/json"}
+    r = requests.post(zenodo_url+'/api/deposit/depositions',
+                   params={'access_token':zenodo_api_key}, json={},
+                   headers=headers)
+    if r.status_code != 201:
+        print("Failure of zenodo record creation. Terminating")
+        sys.exit(1)
+    else:
+        newDOI = r.json()['metadata']['prereserve_doi']['doi']
+        print("Empty record created with DOI:"+newDOI)
+        #print(r.json())
+
+    #populate record
+    deposition_id = r.json()['id']
+    r = requests.put(zenodo_url+'/api/deposit/depositions/%s' % deposition_id,
+                  params={'access_token': zenodo_api_key}, data=json.dumps(sasview_data),
+                  headers=headers)
+    if r.status_code != 200:
+        print("Failure to set metadata on record. Terminating")
+        sys.exit(1)
+    else:
+        print("Record metadata for DOI %s updated" % newDOI)
+
+
+    #next, manually add necessary files from Github and press the publish button
+
+    print("Record has been created on Zenodo with DOI: "+newDOI+"\nNext:\n- Add DOI to appropriate places in SasView code\n- "
+                                                            "Make GitHub release\n - Download all release files from GitHub "
+                                                            "and attach to Zenodo release record.\n- Publish Zenodo record")
+    return newDOI
+
+def update_sasview_init(version, doi):
+    """
+
+    :return:
+    """
+
+    init_file = os.path.join('sasview','src', 'sas', 'sasview', '__init__.py')
+    output_lines = []
+    year = datetime.datetime.now().year
+    with open(init_file, 'r') as f:
+        for line in f.readlines():
+            if line[:11] == '__version__':
```

Comment:
This probably would be safer if we didn't count on strings being at the start of the line?
So instead
`if line[:11] == '__version__'`
we could have
`if '__version__' in line`:

similarly for all other tests